src/jbuild.ml

```diff
@@ -1114,21 +1164,26 @@ module Executables = struct
         (t, Some { Install_conf. section = Bin; files; package })
 
   let public_name =
-    string >>| function
+    located string >>| fun (loc, s) ->
+    (loc
+    , match s with
     | "-" -> None
-    | s   -> Some s
+    | s   -> Some s)
 
   let multi =
     record
       (let%map names, public_names =
          map_validate
-           (let%map names = field "names" (list (located string))
+           (let%map names = field_o "names" (list (located string))
             and pub_names = field_o "public_names" (list public_name) in
             (names, pub_names))
            ~f:(fun (names, public_names) ->
-             match public_names with
-             | None -> Ok (names, List.map names ~f:(fun _ -> None))
-             | Some public_names ->
+             match names, public_names with
+             | None, None -> Ok ([], [])
+             | None, Some p -> Ok ([], p)
```

Comment:
It seems to me that it would be clearer to keep these values as options here. In particular, with the current code do we get the right error message if we write:

```
(executables
 (names))
```

?

Reply:
I thought about this before this PR and I actually think that this case is best handled at the sexp layer. I think that adding a primitive like `list1` would be better here and would handle other cases as well.

Reply:
adding list1 seems fine. However i still think keeping options here would make the program logic more explicit. In `common` you are matching on whether both lists are empty or not, which means checking whether the fields were absent or present. Using options makes this fact explicit. Otherwise you have to remember that "empty list" means "field is absent".

Reply:
Okay, I switched both names and public_names to an option type.